kernel/src/common/deferred_call.rs

```diff
@@ -23,21 +23,21 @@ struct AtomicUsize {
 }
 
 impl AtomicUsize {
```

Comment:
Should this be crate-restricted? I could imagine cases when peripherals (e.g., that directly handle interrupts) might need it. Right now we put all interrupts through kernel serialization but it's important that chips be able to do this in theory. Also, depending on the memory consistency model, they might need to.

Reply:
You're right, though #1083 proposes getting rid of this anyway since there now there is a replacement for it in Rust's core library (which is like what other code should use).